Components/webComponents/cartComponents/bigCartComponent.js

```diff
@@ -1,82 +1,77 @@
-import { CartApi } from "../../../API/CartApi.js";
-
 export class BigCart extends HTMLElement {
     constructor() {
         super();
-        this.items = [];
+        this.renderedItems = [];
     }
 
     async connectedCallback() {
         this.innerHTML = `
             <div id="cart-items"></div>
         `;
+    }
+
+    static get observedAttributes() {
+        return ["items"];
+    }
 
-        let cartApi = new CartApi();
-        let cart = await cartApi.fetchCart();
-        this.items.push(cart.items);
-        console.log(this.items);
-        this.renderCartBigCart(cart.items);
+    attributeChangedCallback(name, oldValue, newValue) {
+        if (name === "items") {
+            this.render(JSON.parse(newValue))
+        }
+    }
+    
+    render(allItemsInState) {
+        const newItems = allItemsInState.filter(itemInState => this.isNewItem(itemInState));
+        const updatedItems = allItemsInState.filter(itemInState => this.isUpdatedItem(itemInState));
+        const deletedItems = this.renderedItems.filter(renderedItem => this.hasBeenDeletedFrom(renderedItem, allItemsInState));
```

Comment:
What will be the content of `newItems`, `updatedItems` and `deletedItems`?

Reply:
@kristandre , `newItems` will be a list of all the items inside cart which is checked against the local "state". If the item is not in the local state it will return false.
`updatedItems` returns a list and checks each item to see if there has been a increase or decrease in amount. If there has been it will return true.
`deletedItems` takes the rendered items and checks if an item in state is not there against the items in cart.